java/statestore-commit/src/main/java/sleeper/commit/StateStoreCommitRequestJson.java

```diff
@@ -44,25 +48,37 @@ public StateStoreCommitRequestJson(CommitRequestType type, CommitRequest request
     public StateStoreCommitRequest getCommitRequest() {
         if (CommitRequestType.COMPACTION_FINISHED == type) {
             return StateStoreCommitRequest.forCompactionJob(request.toCompactionJobCommitRequest());
-        } else {
-            throw new CommitRequestValidationException("Commit request type not recognised: " + type);
         }
+        if (CommitRequestType.INGEST_ADD_FILES == type) {
+            return StateStoreCommitRequest.forIngestJob(request.toIngestJobCommitRequest());
+        }
+        throw new CommitRequestValidationException("Commit request type not recognised: " + type);
     }
 
     /**
      * Represents a compaction job commit request.
      */
     private static class CommitRequest {
```

Comment:
It might be worth changing which class we deserialise to based on the type, so we don't need to put all the potential fields here.